### DISCUSSION TITLE
Window always on top

### DISCUSSION THREAD

**Discussion by vxdy** [slint-ui/slint#5261] [Q&A]
Is it possible to have the Window created by slint always on top? so when i click on another program, i can still see the Slint Window?

**Comment by Vadoola** [comment]
The main Window Component from Slint has an "always-on-top" property that you can use. It doesn't work on Linux+Wayland, but it works with MS Windows, and X11. I don't have a Mac so I don't know if it works on MacOS.

https://releases.slint.dev/1.6.0/docs/slint/src/language/builtins/elements.html#window

**Reply by Gibbz** [reply]
Will Wayland be supported soon?

**Reply by Vadoola** [reply]
As far as I'm aware there is no Wayland protocol for supporting this from an App. I think I saw somewhere that it was believed always on top support was the job of the Window Manager not individual applications (I may be wrong here). If Wayland doesn't support the feature, Slint won't be able to either.

**Reply by james-tharit** [reply]
I can confirm that it's working on Mac

**Comment by Gibbz** [comment]
There is this I'm trying to use with slint, but I don't think it's the same thing...

https://wayland.app/protocols/wlr-layer-shell-unstable-v1

**Reply by Vadoola** [reply]
As I understand it those are the APIs you would use if you were trying to write a shell or compositor, not an application. That being said I'm honestly out of my depth here, I really don't know much about directly interacting with Wayland, and havent looked closely, so I may be completely wrong.

**Reply by Gibbz** [reply]
Okay. Im doing some research into it. Will report back anything if it's of use 👍
